diff --git a/build.gradle b/build.gradle
index a677276..262357d 100644
--- a/build.gradle
+++ b/build.gradle
@@ -70,6 +70,7 @@ subprojects {
                   path = project.protoc
                 } else {
                   artifact = "com.google.protobuf:protoc:${protobufVersion}"
+                  path = '/usr/bin/protoc'
                 }
               }
               plugins {
diff --git a/compiler/build.gradle b/compiler/build.gradle
index 2302e5c..d67adfc 100644
--- a/compiler/build.gradle
+++ b/compiler/build.gradle
@@ -59,11 +59,17 @@ model {
     x86_64 {
       architecture "x86_64"
     }
+    ppc64le {
+      architecture "ppc64le"
+    }
+    s390x {
+      architecture "s390x"
+    }
   }
 
   components {
     java_plugin(NativeExecutableSpec) {
-      if (arch in ['x86_32', 'x86_64']) {
+      if (arch in ['x86_32', 'x86_64', 'ppc64le', 's390x']) {
         // If arch is not within the defined platforms, we do not specify the
         // targetPlatform so that Gradle will choose what is appropriate.
         targetPlatform arch
diff --git a/compiler/check-artifact.sh b/compiler/check-artifact.sh
index a915e2a..efa449d 100755
--- a/compiler/check-artifact.sh
+++ b/compiler/check-artifact.sh
@@ -59,6 +59,10 @@ checkArch ()
         assertEq $format "elf32-i386" $LINENO
       elif [[ "$ARCH" == x86_64 ]]; then
         assertEq $format "elf64-x86-64" $LINENO
+      elif [[ "$ARCH" == ppc64le ]]; then
+        assertEq $format "elf64-ppc64le" $LINENO
+      elif [[ "$ARCH" == s390x ]]; then
+        assertEq $format "elf64-s390" $LINENO
       else
         fail "Unsupported arch: $ARCH"
       fi
@@ -101,7 +105,7 @@ checkDependencies ()
     dump_cmd='ldd '"$1"
     if [[ "$ARCH" == x86_32 ]]; then
       white_list="linux-gate\.so\.1\|libpthread\.so\.0\|libm\.so\.6\|libc\.so\.6\|ld-linux\.so\.2"
-    elif [[ "$ARCH" == x86_64 ]]; then
+    elif [[ "$ARCH" == x86_64 || "$ARCH" == ppc64le || "$ARCH" == s390x ]]; then
       white_list="linux-vdso\.so\.1\|libpthread\.so\.0\|libm\.so\.6\|libc\.so\.6\|ld-linux-x86-64\.so\.2"
     fi
   elif [[ "$OS" == osx ]]; then
